src/channel/Channel.js

```diff
@@ -244,11 +251,10 @@ export function decodeUnaryResponse(data) {
     // 0 = successful
     let status = 0;
 
-    while (dataOffset < data.byteLength) {
-        const dataView = new DataView(data, dataOffset);
-        const frameByte = dataView.getUint8(0);
+    while (dataOffset < dataView.byteLength) {
+        const frameByte = dataView.getUint8(dataOffset + 0);
         const frameType = frameByte >> 7;
-        const frameByteLength = dataView.getUint32(1);
+        const frameByteLength = dataView.getUint32(dataOffset + 1);
         const frameData = new Uint8Array(data, dataOffset + 5, frameByteLength);
```

Comment:
Upon a recheck, this is still not guarded here on this line, as `data` is an `ArrayBuffer` instance.
I'll file a follow-up PR.